src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -944,25 +944,42 @@ internal bool VerifyRemoteCertificate(RemoteCertificateValidationCallback? remot
                 else
                 {
                     chain ??= new X509Chain();
-                    chain.ChainPolicy.RevocationMode = _sslAuthenticationOptions.CertificateRevocationCheckMode;
-                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
 
-                    // Authenticate the remote party: (e.g. when operating in server mode, authenticate the client).
-                    chain.ChainPolicy.ApplicationPolicy.Add(_sslAuthenticationOptions.IsServer ? s_clientAuthOid : s_serverAuthOid);
-
-                    if (trust != null)
+                    if (_sslAuthenticationOptions.ValidationPolicy != null)
                     {
-                        chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
-                        if (trust._store != null)
+                        chain.ChainPolicy = _sslAuthenticationOptions.ValidationPolicy;
+                        if (_sslAuthenticationOptions.ValidationPolicy.VerificationTimeIgnored)
                         {
-                            chain.ChainPolicy.CustomTrustStore.AddRange(trust._store.Certificates);
+                            // Update VerificationTime to 'Now' unless explicitly set for whatever reason.
+                            chain.ChainPolicy.VerificationTime = DateTime.Now;
                         }
```

Comment:
This should be an implementation detail of X509Chain, not edited here.

Reply:
https://github.com/dotnet/runtime/blob/c9feedf55ea613a2b9220fe6b9e8feda2aa72287/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509Chain.cs#L136

Change

```diff
-chainPolicy.VerificationTime,
+chainPolicy.VerificationTimeIgnored ? DateTime.Now : VerificationTime,
```